From: Hauke Mehrtens <hauke@hauke-m.de>
Date: Mon, 24 Feb 2020 00:00:00 +0100
Subject: [PATCH] mac80211: Allow IBSS mode and different beacon intervals

ath10k-ct supports the combination to select IBSS (ADHOC) mode and
different beacon intervals together. mac80211 does not like this
combination, but Ben says this is ok, so remove this check.

ath10k-ct starting with version 5.2 allows the combination of
NL80211_IFTYPE_ADHOC and beacon_int_min_gcd in ath10k_10x_ct_if_comb
which triggers this warning. Ben told me that this is not a big problem
and we should ignore this.
---
 net/wireless/core.c | 15 ---------------
 1 file changed, 15 deletions(-)

--- a/net/wireless/core.c
+++ b/net/wireless/core.c
@@ -614,21 +614,6 @@ static int wiphy_verify_combinations(str
 				    c->limits[j].max > 1))
 				return -EINVAL;
 
-			/*
-			 * This isn't well-defined right now. If you have an
-			 * IBSS interface, then its beacon interval may change
-			 * by joining other networks, and nothing prevents it
-			 * from doing that.
-			 * So technically we probably shouldn't even allow AP
-			 * and IBSS in the same interface, but it seems that
-			 * some drivers support that, possibly only with fixed
-			 * beacon intervals for IBSS.
-			 */
-			if (WARN_ON(types & BIT(NL80211_IFTYPE_ADHOC) &&
-				    c->beacon_int_min_gcd)) {
-				return -EINVAL;
-			}
-
 			cnt += c->limits[j].max;
 			/*
 			 * Don't advertise an unsupported type
